src/components/enterprise-page/data/service.js

```diff
@@ -22,3 +23,28 @@ export const updateUserActiveEnterprise = (enterpriseId) => {
     formData,
   );
 };
+
+const fetchEnterpriseLearnerData = async (options) => {
+  const config = getConfig();
+  const enterpriseLearnerUrl = `${config.LMS_BASE_URL}/enterprise/api/v1/enterprise-learner/`;
+  const queryParams = new URLSearchParams({
+    ...options,
+    page: 1,
+  });
+  const url = `${enterpriseLearnerUrl}?${queryParams.toString()}`;
+  const response = await getAuthenticatedHttpClient().get(url);
+  const data = camelCaseObject(response.data);
+  let results = data.results;
+  let next = data.next;
+  while (next) {
+    const nextPage = await getAuthenticatedHttpClient().get(next);
+    const nextData = camelCaseObject(nextPage.data);
+    next = nextData.next;
+    results = results.concat(nextData.results);
+  }
```

Comment:
This is likely sufficient for this use case!

Just a heads up, React Query does have support for `useInfiniteQuery` ([source](https://tanstack.com/query/latest/docs/react/reference/useInfiniteQuery)) which appears to have some helpers for traversing paginated responses. Not sure how applicable it is to this use case, though.